### ISSUE TITLE
Type checking/violation for tests when checking using setExpectedException

### ISSUE THREAD

**Issue by mwjames** [scrutinizer-ci/php-analyzer#597]
> Here, you can explain why the found issue was invalid, and what result
> you would have expected otherwise.
>
> Please also check that this issue has not already been reported.

see https://scrutinizer-ci.com/g/onoi/event-dispatcher/inspections/c9a73dc3-3be7-485f-9221-b7f081e630c5/issues/files/tests/phpunit/Unit/Listener/GenericEventListenerCollectionTest.php?status=new&orderField=path&order=asc&issueId=8dfaa054-d3f5-4d0e-92e5-d461eb0e7c1b

The test checks for an exception but

```
	$this->setExpectedException( 'RuntimeException' );
	$instance->registerCallback( 'foo', new \stdClass );
```

scrutinizer-ci is reporting `new \stdClass() is of type object<stdClass>, but the function expects a callable.` as type violation which normally would be correct but the intention of the test to check the violation using `setExpectedException` and therefore the assumption about the type is misplaced.
